prinz/src/main/scala/pl/touk/nussknacker/prinz/model/repository/CompositeModelRepository.scala

```diff
@@ -0,0 +1,34 @@
+package pl.touk.nussknacker.prinz.model.repository
+import pl.touk.nussknacker.prinz.model.{Model, ModelName}
+
+class CompositeModelRepository private(repositories: List[ModelRepository]) extends ModelRepository {
+
+  override def listModels: RepositoryResponse[List[Model]] =
+    repositories.map(_.listModels).partition(_.isLeft) match {
+      case (Nil, lists) => Right((for (Right(list) <- lists) yield list).flatten)
+      case(exceptions,  _) => Left((for (Left(exc) <- exceptions) yield exc).head)
+    }
+
+  override def getModel(name: ModelName): RepositoryResponse[Model] = {
+    val firstRepository = repositories.head
+    repositories.drop(1).foldLeft(firstRepository.getModel(name))(accumulateFirstAvailable(name))
+  }
```

Comment:
This way of getting the model seems complicated Wouldnt it be more readable to do it iteratively or using the find function?

Reply:
done